controller/src/main/java/org/jboss/as/controller/ValidateModelStepHandler.java

```diff
@@ -113,5 +161,22 @@ boolean hasAlternative(final String[] alternatives, ModelNode operationObject) {
         return false;
     }
 
+    private Resource loadResource(OperationContext context) {
+        final PathAddress address = context.getCurrentAddress();
+        PathAddress current = PathAddress.EMPTY_ADDRESS;
+        Resource resource = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS, false);
+        for (PathElement element : address) {
```

Comment:
To be more forgiving if a resource disappears for whatever reason. The thing that led me to this was finding that a composite to
    add a=a
    remove a=b
would cause an exception failing validation. Also exceptions are very expensive to create, so I'm just playing safe.